cmp/grid/GridStateModel.js

```diff
@@ -189,9 +191,24 @@ export class GridStateModel {
     }
 
     //--------------------------
-    // Helper
+    // Helpers
     //--------------------------
     saveStateChange = debounce(() => {
         this.saveState(this.getStateKey(), this.state);
     }, GridStateModel.STATE_SAVE_DEBOUNCE_MS);
+
+    /**
+     * Return a copy of the passed-in columnState with the width removed from non-resizable columns.
+     * Does not mutate the input.
+     * @param columnState The column state to be processed.
+     * @returns A copy of columnState with width removed from non-resizable columns.
+     */
+    removeWidthOfResizable(columnState) {
+        const cols = this.gridModel.getLeafColumns();
+
+        return columnState.map(state => {
+            const col = this.gridModel.findColumn(cols, state.colId);
+            return col.resizable ? state : omit(state, 'width');
+        });
+    }
```

Comment:
A few small comments here:
* Believe this method name is backwards - we are stripping width from *non*-resizable cols 😄 
* But a more general name would be better anyway - `preprocessColumnState` or `cleanColumnState`.  It could then both remove any non-existent cols and strip the widths. That would have the added benefit of it ensuring it was internally defending against missing columns vs. relying on the caller calling it after stripping null cols.
* I use the full jsdoc comment style heavily in code I write (as you might have noticed...) but only for public methods on a class. Our standard is to avoid it for implementation comments, precisely because it makes them look like public APIs. (I also primarily add those docs for the types, FWIW, which we don't have set here anyway.)

Reply:
I have updated this to reflect your comments.